### PR TITLE
Format expiry to work with foreign keyboards

### PR DESCRIPTION
**Pull request by ajkelso**
This PR will change the eventListener from `keypress` to `beforeinput` in order to accommodate for foreign keyboards on macs that bypass the `keypress` event.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -584,10 +588,10 @@
         month = el[0], year = el[1];
         this.formatCardExpiryMultiple(month, year);
       } else {
-        QJ.on(el, 'keypress', restrictCombinedExpiry);
-        QJ.on(el, 'keypress', formatExpiry);
-        QJ.on(el, 'keypress', formatForwardSlash);
-        QJ.on(el, 'keypress', formatForwardExpiry);
```
Comment (chrissrogers) [inline lib/index.js:590]: We'll need to retain support for the `keypress` in addition to supporting the `beforeinput` event as `beforeinput` is [not supported in IE11](https://caniuse.com/?search=beforeinput), which we still need to support for now.

Comment (ajkelso) [reply lib/index.js:590]: Done.

File: spec/index.spec.coffee [hunk 2]
```diff
@@ -516,9 +516,9 @@ describe 'payment', ->
       Payment.formatCardExpiry(expiry)
 
       ev = document.createEvent "HTMLEvents"
-      ev.initEvent "keypress", true, true
-      ev.eventName = "keypress"
-      ev.which = "/".charCodeAt(0)
+      ev.initEvent "beforeinput", true, true
+      ev.eventName = "beforeinput"
+      ev.data = "/"
```
Comment (chrissrogers) [inline spec/index.spec.coffee]: Let's replicate the tests so that we cover both `keypress` and `beforeinput`

Comment (ajkelso) [reply spec/index.spec.coffee]: Done.

**Pull request merged**